### PR TITLE
DSD-891: Adding a11y docs for page layout components

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-891](https://jira.nypl.org/browse/DSD-891)

## This PR does the following:

- Adds table of contents and accessibility docs for `HorizontalRule`, `SimpleGrid`, `StructuredContent`, `Table`, and `Template`.

## How has this been tested?

Storybook.

## Accessibility concerns or updates

- More docs.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr954-xlskj5aapcjqmeffwyyjr1zxdlidl4hh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/62586d68bda4981539f77f49

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr954-xlskj5aapcjqmeffwyyjr1zxdlidl4hh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/62586d68bda4981539f77f49

File: src/components/Template/Template.stories.mdx [hunk 1]
```diff
@@ -65,10 +65,10 @@ import { getCategory } from "../../utils/componentCategories";
 ## Table of Contents
 
 - [Accessibility for TemplateAppContainer](#accessibility-for-templateappcontainer)
```
Comment (bigfishdesign13) [inline src/components/Template/Template.stories.mdx]: Is `Accessibility for TemplateAppContainer` the first item in the TOC for a reason?  Or should it be moved down below `TemplateAppContainer Component Props` and renamed to `TemplateAppContainer Accessibility`?

**Review by bigfishdesign13 — COMMENTED** [review]
Overall, looks good, but I had one question.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr954-xlskj5aapcjqmeffwyyjr1zxdlidl4hh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/62586d68bda4981539f77f49

**Comment by EdwinGuzman** [comment]
Updated

**Review by bigfishdesign13 — APPROVED** [review]
Exceptional.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr954-xlskj5aapcjqmeffwyyjr1zxdlidl4hh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/62586d68bda4981539f77f49

**Pull request merged**